ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/CommandDeduplicationSpec.scala

```diff
@@ -44,165 +52,304 @@ class CommandDeduplicationSpec
+        val dedupValue = newDedupValue(
+          _.setRecordTimeBounds(buildPreExecutionDeduplicationBounds(timestamp, timestamp))
+        )
+        val commitContext = createCommitContext(None, Map(aDedupKey -> Some(dedupValue)))
+        commitContext.minimumRecordTime = Some(timestamp.subtract(Duration.ofMillis(1)))
+        commitContext.maximumRecordTime = Some(timestamp)
 
-      actual match {
-        case StepContinue(_) => succeed
-        case StepStop(_) => fail()
+        deduplicateStepHasTransactionRejectionEntry(commitContext)
       }
-    }
 
-    "continue if record time is after deduplication time in case a deduplication entry is found" in {
-      val dedupValue = newDedupValue(aRecordTime)
-      val inputs = Map(aDedupKey -> Some(dedupValue))
-      val context =
-        createCommitContext(recordTime = Some(aRecordTime.addMicros(1)), inputs = inputs)
+      "set the out of time bounds log entry during rejections" in {
+        val dedupValue = newDedupValue(
+          _.setRecordTimeBounds(buildPreExecutionDeduplicationBounds(timestamp, timestamp))
+        )
+        val commitContext = createCommitContext(None, Map(aDedupKey -> Some(dedupValue)))
+        commitContext.minimumRecordTime = Some(timestamp)
+        commitContext.maximumRecordTime = Some(timestamp)
 
-      val actual = deduplicateCommandStep(context, aTransactionEntrySummary)
+        deduplicateStepHasTransactionRejectionEntry(commitContext)
+        commitContext.outOfTimeBoundsLogEntry shouldBe 'defined
+      }
 
-      actual match {
-        case StepContinue(_) => succeed
-        case StepStop(_) => fail()
+      "return the command deduplication duration as deduplication duration when this exceeds the max delta between record times" in {
+        val dedupValue = newDedupValue(
+          _.setRecordTimeBounds(buildPreExecutionDeduplicationBounds(timestamp, timestamp))
+        )
+        val commitContext = createCommitContext(None, Map(aDedupKey -> Some(dedupValue)))
+        commitContext.minimumRecordTime = Some(timestamp)
+        commitContext.maximumRecordTime = Some(timestamp)
+
+        val rejectionEntry = deduplicateStepHasTransactionRejectionEntry(commitContext)
+        rejectionEntry.getSubmitterInfo.getDeduplicationDuration shouldBe buildDuration(
+          deduplicationDuration
+        )
       }
-    }
 
-    "produce rejection log entry in case record time is on or before deduplication time" in {
-      for (
-        (recordTime, deduplicationTime) <- Iterable(
-          (aRecordTime, aRecordTime),
-          (aRecordTime, aRecordTime.addMicros(1)),
+      "the deduplication duration is the delta between records when this exceeds the command deduplication rejection" in {
+        val dedupValue = newDedupValue(
+          _.setRecordTimeBounds(buildPreExecutionDeduplicationBounds(timestamp, timestamp))
```

Comment:
When you use the same timestamp for min and max, you can't test in the happy case that the code doesn't inadvertently pick the min record time. How about using two different timestamps here?

Reply:
Good point. I changed the test cases to use different values.